packages/reason-relay/src/ReasonRelay_Internal.re

```diff
@@ -0,0 +1,42 @@
+/**
+ * Various helpers.
+ */
+
+// We occasionally have to remove undefined keys from objects, something I haven't figured out how to do with pure BuckleScript
+let internal_cleanObjectFromUndefinedRaw = [%raw
+  {|
+  function (obj) {
+    var newObj = {};
+
+    Object.keys(obj).forEach(function(key) {
+      if (typeof obj[key] !== 'undefined') {
+        newObj[key] = obj[key];
+      }
+    });
+
+    return newObj;
+  }
+|}
+];
+
+// Since BS compiles unit to 0, we have to convert that to an empty object when dealing with variables in order for Relay to be happy
+let internal_cleanVariablesRaw = [%raw
```

Comment:
I think now BS compiles `unit` to `undefined`.
But wouldn't it be better for cases where there is no variables to type it as `Js.t({.})` instead of unit? Then there would be no need for this function.

Reply:
Yes, this is still there for compat with earlier `bs-platform`, but I also think it could be removed soon.

Reply:
ok! `unit` compiled to `undefined` is here since BS 7.3 if I'm not mistaken.

Reply:
Well in that case we should definitively be good to go! I would still like to make this a separate change in another PR though 😬  Let's follow up when this is merged!